clippy subcommand does not work

```
os: macOS Sierra 10.12.1 (16B2555)

multirust: default toolchain: nightly
multirust: default location: /Users/marek/.multirust/toolchains/nightly

rustc 1.14.0-nightly (cae6ab1c4 2016-11-05)
cargo 0.13.0-nightly (9c47815 2016-11-04)

MULTIRUST_HOME=/Users/marek/.multirust
MULTIRUST_TOOLCHAIN=nightly
```

steps to reproduce

1. Install clippy as cargo subcommand
    ```
    cargo install clippy
    ```
2. Create new project and enter it
    ```
    cargo new empty
    cd empty
    ```
3. Run clippy
    ```
    cargo clippy
    ```

error:

```
warning: dependency (clippy) specified without providing a local path, Git repository, or version to use. This will be considered an error in future versions
   Compiling empty v0.1.0 (file:///Users/marek/projects/ethcore/empty)
     Running `/Users/marek/.multirust/toolchains/nightly/cargo/bin/cargo-clippy src/lib.rs --crate-name empty --crate-type lib -g -L /Users/marek/projects/ethcore/empty/target/debug/deps --sysroot /Users/marek/.multirust/rustup/toolchains/nightly -Zno-trans --cfg feature=\"cargo-clippy\" -C metadata=bed0750e55bcd5de --out-dir /Users/marek/projects/ethcore/empty/target/debug/deps --emit=dep-info,link -L dependency=/Users/marek/projects/ethcore/empty/target/debug/deps`
error[E0463]: can't find crate for `std`

error: aborting due to previous error

thread 'main' panicked at 'Box<Any>', ../src/librustc_errors/lib.rs:461
stack backtrace:
   1:        0x10af1699a - std::sys::imp::backtrace::tracing::imp::write::he3d1bfbdbf113480
   2:        0x10af2675f - std::panicking::default_hook::{{closure}}::h575f1b40d2e88f07
   3:        0x10af2391f - std::panicking::default_hook::h3d5dccce8125d0cf
   4:        0x10af23f56 - std::panicking::rust_panic_with_hook::h00b81bb3dcbd51f2
   5:        0x10adae5db - std::panicking::begin_panic::h6a56eabf4545c581
   6:        0x10adc000c - rustc_errors::Handler::abort_if_errors::hbaaffb7ebee22cf1
   7:        0x107642199 - rustc_metadata::locator::Context::report_errs::h7ca356b288af9547
   8:        0x10763579e - rustc_metadata::creader::CrateLoader::read_extension_crate::h71cda1425d04bd98
   9:        0x10763897c - <rustc_metadata::creader::CrateLoader<'a> as rustc::middle::cstore::CrateLoader>::process_item::h28de7cbe01fcdc2c
  10:        0x106f85b65 - rustc_resolve::build_reduced_graph::<impl rustc_resolve::Resolver<'b>>::build_reduced_graph_for_item::h3d48aa5e276489d0
  11:        0x106f8c572 - <rustc_resolve::build_reduced_graph::BuildReducedGraphVisitor<'a, 'b> as syntax::visit::Visitor>::visit_item::h709efad32d31e782
  12:        0x106f8cb4e - <rustc_resolve::build_reduced_graph::BuildReducedGraphVisitor<'a, 'b> as syntax::visit::Visitor>::visit_item::h709efad32d31e782
  13:        0x106f8123e - rustc_resolve::macros::<impl syntax::ext::base::Resolver for rustc_resolve::Resolver<'a>>::visit_expansion::hb5d29f0018bc1ba5
  14:        0x10aa3d19e - syntax::ext::expand::MacroExpander::collect_invocations::h01df4833c024895b
  15:        0x10aa39276 - syntax::ext::expand::MacroExpander::expand::h1fb862fdd8eea6f2
  16:        0x10aa38b53 - syntax::ext::expand::MacroExpander::expand_crate::h5d27b1f5f31cc0ce
  17:        0x106840bcb - rustc_driver::driver::phase_2_configure_and_expand::{{closure}}::h21a10cc1de20a39a
  18:        0x1067f5628 - rustc_driver::driver::phase_2_configure_and_expand::h56e2afa5f3ad2a15
  19:        0x1067efeae - rustc_driver::driver::compile_input::h21840cdf516c3ab1
  20:        0x10681bd39 - rustc_driver::run_compiler::h81a62653df4e7b03
  21:        0x1064e4855 - cargo_clippy::main::hcd30f9d87e2c270a
  22:        0x10af26d1a - __rust_maybe_catch_panic
  23:        0x10af247e6 - std::rt::lang_start::h74b3afbdd8daef1c
error: Could not compile `empty`.

Caused by:
  process didn't exit successfully: `/Users/marek/.multirust/toolchains/nightly/cargo/bin/cargo-clippy src/lib.rs --crate-name empty --crate-type lib -g -L /Users/marek/projects/ethcore/empty/target/debug/deps --sysroot /Users/marek/.multirust/rustup/toolchains/nightly -Zno-trans --cfg feature="cargo-clippy" -C metadata=bed0750e55bcd5de --out-dir /Users/marek/projects/ethcore/empty/target/debug/deps --emit=dep-info,link -L dependency=/Users/marek/projects/ethcore/empty/target/debug/deps` (exit code: 101)
```

I've also checked if running clippy as optional dependency works. It does

--- a/src/main.rs
+++ b/src/main.rs
@@ -149,23 +149,6 @@ pub fn main() {
 
     let dep_path = env::current_dir().expect("current dir is not readable").join("target").join("debug").join("deps");
 
-    let home = option_env!("RUSTUP_HOME").or(option_env!("MULTIRUST_HOME"));
-    let toolchain = option_env!("RUSTUP_TOOLCHAIN").or(option_env!("MULTIRUST_TOOLCHAIN"));
-    let sys_root = if let (Some(home), Some(toolchain)) = (home, toolchain) {
-        format!("{}/toolchains/{}", home, toolchain)
-    } else {
-        option_env!("SYSROOT")
-            .map(|s| s.to_owned())
-            .or(Command::new("rustc")
-                .arg("--print")
-                .arg("sysroot")
-                .output()
-                .ok()
-                .and_then(|out| String::from_utf8(out.stdout).ok())
-                .map(|s| s.trim().to_owned()))
-            .expect("need to specify SYSROOT env var during clippy compilation, or use rustup or multirust")
-    };
-
     if let Some("clippy") = std::env::args().nth(1).as_ref().map(AsRef::as_ref) {
         // this arm is executed on the initial call to `cargo clippy`
 
@@ -201,11 +184,11 @@ pub fn main() {
             let args = std::env::args().skip(2);
             if let Some(first) = target.kind.get(0) {
                 if target.kind.len() > 1 || first.ends_with("lib") {
-                    if let Err(code) = process(std::iter::once("--lib".to_owned()).chain(args), &dep_path, &sys_root) {
+                    if let Err(code) = process(std::iter::once("--lib".to_owned()).chain(args), &dep_path) {
                         std::process::exit(code);
                     }
                 } else if ["bin", "example", "test", "bench"].contains(&&**first) {
-                    if let Err(code) = process(vec![format!("--{}", first), target.name].into_iter().chain(args), &dep_path, &sys_root) {
+                    if let Err(code) = process(vec![format!("--{}", first), target.name].into_iter().chain(args), &dep_path) {
                         std::process::exit(code);
                     }
                 }
@@ -215,6 +198,23 @@ pub fn main() {
         }
     } else {
         // this arm is executed when cargo-clippy runs `cargo rustc` with the `RUSTC` env var set to itself
+
+        let home = option_env!("RUSTUP_HOME").or(option_env!("MULTIRUST_HOME"));
+        let toolchain = option_env!("RUSTUP_TOOLCHAIN").or(option_env!("MULTIRUST_TOOLCHAIN"));
+        let sys_root = if let (Some(home), Some(toolchain)) = (home, toolchain) {
+            format!("{}/toolchains/{}", home, toolchain)
+        } else {
+            option_env!("SYSROOT")
+                .map(|s| s.to_owned())
+                .or(Command::new("rustc")
+                    .arg("--print")
+                    .arg("sysroot")
+                    .output()
+                    .ok()
+                    .and_then(|out| String::from_utf8(out.stdout).ok())
+                    .map(|s| s.trim().to_owned()))
+                .expect("need to specify SYSROOT env var during clippy compilation, or use rustup or multirust")
+        };
 
         // this conditional check for the --sysroot flag is there so users can call `cargo-clippy` directly
         // without having to pass --sysroot or anything
@@ -239,7 +239,7 @@ pub fn main() {
     }
 }
 
-fn process<P, I>(old_args: I, dep_path: P, sysroot: &str) -> Result<(), i32>
+fn process<P, I>(old_args: I, dep_path: P) -> Result<(), i32>
     where P: AsRef<Path>,
           I: Iterator<Item = String>
 {
@@ -256,8 +256,6 @@ fn process<P, I>(old_args: I, dep_path: P, sysroot: &str) -> Result<(), i32>
     }
     args.push("-L".to_owned());
     args.push(dep_path.as_ref().to_string_lossy().into_owned());
-    args.push(String::from("--sysroot"));
-    args.push(sysroot.to_owned());
     args.push("-Zno-trans".to_owned());
     args.push("--cfg".to_owned());
     args.push(r#"feature="cargo-clippy""#.to_owned());

